linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py

```diff
@@ -0,0 +1,93 @@
+import sys
+import time
+
+import numpy as np
+
+matrixab = np.loadtxt("matrix.txt")
+B = np.copy(matrixab[:, matrixab.shape[1] - 1])
+
+
+def foo(matrix):
+    start = time.process_time()
+    ab = np.copy(matrix)
+    numofrows = ab.shape[0]
+    numofcolumns = ab.shape[1] - 1
+    xlst = []
+
+    """Lead element search"""
+    print("Matrix before leading coefficient search: ")
+    print(ab)
+    print(" ")
+
+    """Upper triangular matrix"""
+    for columnnum in range(numofrows):
+        for i in range(columnnum, numofcolumns):
+            if abs(ab[i][columnnum]) > abs(ab[columnnum][columnnum]):
+                ab[[columnnum, i]] = ab[[i, columnnum]]
+                if ab[columnnum, columnnum] == 0.0:
+                    sys.exit("Matrix is not correct")
+            else:
+                pass
+        if columnnum != 0:
+            for i in range(columnnum, numofrows):
+                ab[i, :] -= (
+                    ab[i, columnnum - 1]
+                    / ab[columnnum - 1, columnnum - 1]
+                    * ab[columnnum - 1, :]
+                )
+
+    print("Upper triangular matrix: ")
+    print(ab.round(3))
+    print(" ")
+
+    """Find x vector"""
+    columnnum = numofrows
+    while columnnum != 0:
+        columnnum -= 1
+        lineofx = ab[columnnum, numofrows]
+        if columnnum + 1 != numofrows:
+            for y in range(1, numofrows - columnnum):
+                lineofx += -ab[columnnum, numofrows - y] * xlst[y - 1]
+        x = lineofx / ab[columnnum, columnnum]
+        xlst.append(x)
+
+    stop = time.process_time()
+    xlst.reverse()
+    print("x vector: ")
+    print(xlst)
+    print(" ")
+    print(f"Start time: {start}, End time: {stop}")
+    print(f"Elapsed time during the whole function in seconds: {stop - start}")
+
+    return np.asarray(xlst)
+
+
+vectorofxalpha = foo(matrixab)
```

Comment:
```suggestion
if __name__ == "__main__":
    vectorofxalpha = foo(matrixab)
```
And indent all lines below in this file.